### PR TITLE
Revert "Check root cause for errors"

### PR DESCRIPTION
**Pull request by henriknyman**
This reverts commit f38aa5b488e72c39144c3658d63e932f53f7fa7f.

The reverted commit changed external behaviour in a way that information
from exceptions thrown from procedures or user-defined functions was lost.

### CODE REVIEW

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/ProcedureInteractionTestBase.java [hunk 1]
```diff
@@ -104,10 +104,7 @@
 
 public abstract class ProcedureInteractionTestBase<S>
 {
-    static final String PROCEDURE_TIMEOUT_ERROR = "The transaction has been terminated. Retry your operation in a new " +
-                                                  "transaction, and you should see a successful result. The transaction " +
-                                                  "has not completed within the specified timeout. You may want to retry " +
-                                                  "with a longer timeout. ";
+    static final String PROCEDURE_TIMEOUT_ERROR = "Procedure got: Transaction guard check failed";
```
Comment (pontusmelke) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/ProcedureInteractionTestBase.java:107]: Isn't this a slightly breaking change?

**Review by pontusmelke — CHANGES_REQUESTED** [review]
Looks good, just a question regarding the error message

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/ProcedureInteractionTestBase.java (hunk 1 above)
Comment (henriknyman) [reply enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/ProcedureInteractionTestBase.java:107]: This is what was there before, so it is more like the result of fixing(/un-breaking) a breaking change by reverting it.

Comment (pontusmelke) [reply enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/ProcedureInteractionTestBase.java:107]: 👍

**Review by pontusmelke — APPROVED** [review]

**Pull request merged**